Deploy our OCI artifact to a private registry

chouquette opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* Both Contributor and Reviewer Checklists are available at https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#pull-requests.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
-->
### What does this PR do?

This PR starts deploying our OCI artifacts to an internal registry.

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

This is probably a bit too `datadog-agent` centric and should be reworked to have a base job that handles multiple flavors, but we'll see about that when we get there

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

chouquette committed:
deploy oci image to an internal registry

chouquette committed:
start publishing the actual pipeline artifact

BaptisteFoy commented on .gitlab/deploy_packages/deploy_common.yml:
```diff
@@ -79,12 +79,22 @@ deploy_packages_oci:
+    - git clone https://github.com/DataDog/datadog-packages /tmp/datadog-packages
+    - cd /tmp/datadog-packages/cmd/datadog-package
+    - go build .
+    - ./datadog-package push registry.ddbuild.io/ci/datadog-agent/datadog-agent-ociru:${VERSION} ${OMNIBUS_PACKAGE_DIR}/datadog-agent-${MAJOR_VERSION}.*.oci.tar
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 Suggestion:** ‏Can we use `registry.ddbuild.io/ci/remote-updates/datadog-agent:${VERSION}`? That'd match the tracers

chouquette replied:
For `registry.ddbuild.io/ci/remote-updates/` no problem at all on my end.

Regarding the image name itself, I believe @vboulineau wanted to have a more descriptive name than `datadog-agent`.
The prefix/folder might be enough to disambiguate what kind of `datadog-package` though

BaptisteFoy replied:
True, the name may change again, but it's easy to change.

chouquette committed:
update image name

chouquette committed:
use the same version pattern as other packages

this way we won't need to fiddle with the version on the release
management side, we can just use it as is

chouquette commented:
/merge

amenasria reviewed (approved):
LGTM for Agent CI Exp owned file !
